cmd/clusterctl/client/repository/repository_github.go

```diff
@@ -247,13 +269,24 @@ func (g *gitHubRepository) getReleaseByTag(tag string) (*github.RepositoryReleas
 
 	client := g.getClient()
 
-	release, _, err := client.Repositories.GetReleaseByTag(context.TODO(), g.owner, g.repository, tag)
-	if err != nil {
-		return nil, g.handleGithubErr(err, "failed to read release %q", tag)
-	}
+	var release *github.RepositoryRelease
+	var retryError error
+	_ = wait.PollImmediate(retryableOperationInterval, retryableOperationTimeout, func() (bool, error) {
+		var getReleasesErr error
+		release, _, getReleasesErr = client.Repositories.GetReleaseByTag(context.TODO(), g.owner, g.repository, tag)
+		if getReleasesErr != nil {
+			retryError = g.handleGithubErr(getReleasesErr, "failed to read release %q", tag)
+			return false, nil
+		}
```

Comment:
Suggestion:
If the `getReleasesErr` is a `github.RateLimitError` then we might want to just return the error immediately. If we already hit the rate limiting then no point in retrying again.